File: lib/data/models/repository_model.dart
import '../../domain/entities/repository.dart';
import '../database/app_database.dart';

/// 仓库数据模型 - Entity ↔ Data 转换
extension RepositoryDataExtension on RepositoryData {
  /// 转换为领域实体
  Repository toEntity() {
    return Repository(
      id: id,
      name: name,
      icon: icon ?? 'folder',
      color: color ?? '#3B82F6',
      isArchived: isArchived,
      isDeleted: isDeleted,
      createdAt: createdAt,
      updatedAt: updatedAt,
    );
  }
}

/// 仓库实体扩展 - 转换为 drift 数据
extension RepositoryEntityExtension on Repository {
  /// 转换为 drift 数据对象
  RepositoryData toData() {
    return RepositoryData(
      id: id,
      name: name,
      icon: icon,
      color: color,
      isArchived: isArchived,
      isDeleted: isDeleted,
      createdAt: createdAt,
      updatedAt: updatedAt,
    );
  }
}

File: web/src/data/models/repository-model.ts
import { Repository } from '../../domain/entities/repository';

export interface RepositoryRecord {
  id: string;
  name: string;
  icon: string;
  color: string;
  isArchived: number;
  isDeleted: number;
  createdAt: string;
  updatedAt: string;
}

export const RepositoryModel = {
  toRecord(entity: Repository): RepositoryRecord {
    return {
      id: entity.id,
      name: entity.name,
      icon: entity.icon,
      color: entity.color,
      isArchived: entity.isArchived ? 1 : 0,
      isDeleted: entity.isDeleted ? 1 : 0,
      createdAt: entity.createdAt.toISOString(),
      updatedAt: entity.updatedAt.toISOString(),
    };
  },

  toEntity(record: RepositoryRecord): Repository {
    return new Repository({
      id: record.id,
      name: record.name,
      icon: record.icon,
      color: record.color,
      isArchived: record.isArchived === 1,
      isDeleted: record.isDeleted === 1,
      createdAt: new Date(record.createdAt),
      updatedAt: new Date(record.updatedAt),
    });
  },
};
